tests/integration/main_test.go

```diff
@@ -102,6 +104,16 @@ func ConfigureTestEnv(testenv env.Environment) {
 	) {
 		testenv.AfterEachTest(f)
 	}
+
+	// Teardown
+	// TODO: Uninstall the Helm Chart here as well
+	testenv.Finish(envfuncs.DeleteNamespace(openTelemetryOperatorNamespaces[0]))
+	testenv.Finish(envfuncs.DeleteNamespace(openTelemetryOperatorNamespaces[1]))
+	testenv.Finish(envfuncs.DeleteNamespace(receiverMockNamespace))
+	testenv.Finish(func(ctx context.Context, envConf *envconf.Config) (context.Context, error) {
+		namespace := ctxopts.Namespace(ctx)
+		return envfuncs.DeleteNamespace(namespace)(ctx, envConf)
+	})
```

Comment:
why not just
testenv.Finish(envfuncs.DeleteNamespace(ctxopts.Namespace(ctx)))

Reply:
Because you need to get the namespace from the context, and we get the context only at runtime. Your suggestion acts as if `ctx` is available in the function we're setting these Finish actions in, and it's not.

Reply:
uch, ok